File: JavaScript/Math-Array-String/51.js
// 2864. Maximum Odd Binary Number


/**
 * @param {string} s
 * @return {string}
 */
var maximumOddBinaryNumber = function(s) {
    // Count the number of '1's and '0's in the string
    const ones = s.split('1').length - 1;
    const zeros = s.split('0').length - 1;
    
    // Construct the result:
    // - Place (ones - 1) '1's at the beginning
    // - Place all '0's in the middle
    // - Place the last '1' at the end to make it odd
    return '1'.repeat(ones - 1) + '0'.repeat(zeros) + '1';
};

// Example usage
console.log(maximumOddBinaryNumber("010"));  // Output: "001"
console.log(maximumOddBinaryNumber("0101")); // Output: "1001"

File: Python/Math-Array-String/51.py
# 2864. Maximum Odd Binary Number

class Solution:
    def maximumOddBinaryNumber(self, s: str) -> str:
        # Count the number of 1s and 0s in the string
        ones = s.count('1')
        zeros = s.count('0')
        
        # To form the maximum number, place all the '1's (except one) in the front
        # The last digit must be '1' to make the number odd
        # Place all '0's in between
        return '1' * (ones - 1) + '0' * zeros + '1'
    

sol = Solution()
output = sol.maximumOddBinaryNumber("010")
# output = sol.maximumOddBinaryNumber("000011010")
print(output) # Output: 001